include/superior_mysqlpp/types/array.hpp

```diff
@@ -170,7 +197,7 @@ namespace SuperiorMySqlpp
 
         constexpr auto maxSize() const
         {
-            return array.max_size();
+            return array.max_size() - int(TerminatingZero);
```

Comment:
What about this way?
```suggestion
            return N;
```